wxt_cidrbot/dynamo_api_handler.py

```diff
@@ -54,14 +54,47 @@ def __init__(self):
             logging.error("Environment variable REGION_NAME must be set")
             sys.exit(1)
 
-        self.dynamodb = ""
+        self.dynamodb = None
         self.table = ""
         self.repo_schema_name = '#repo'
         self.reponame_schema = '#reponame'
+        self.boto3_session = None
+        self.boto3_client = None
+        self.current_table_name = None
 
     def get_dynamo(self):
-        self.dynamodb = boto3.resource('dynamodb')
-        self.table = self.dynamodb.Table(self.db_room_name)
+        """
+        if not already done, creates a dynamodb session and switches the current table to the room table
+        
+        return: None
+        """
+        if self.dynamodb is None:
+            self.dynamodb = boto3.resource('dynamodb')
+        if self.current_table_name != self.db_room_name:
+            self.table = self.dynamodb.Table(self.db_room_name)
```

Comment:
if you are using different tables for different things, they should each likely have their own variable or entry in a dict, something like that